### PR TITLE
Replace react-instantsearch-dom with react-instantsearch

### PR DESCRIPTION
**Pull request by e-kr**
React-instantsearch-dom has been deprecated, so I have replaced it by react-instantsearch as recommended. It seems to be working exactly as before, on both big and smaller screens, but let me know if you see something I have missed! I also see the completed searches in the analytics page in the algolia dashboard. 

I also removed react-autosuggest as it was not needed anymore. However it turned out that react-instantsearch itself is not really made for the type of dropdown search list, which caused some problems. I read on an algolia forum that it was recommended to use algolia autocomplete along with react-instantsearch in such cases, which is why i ended up using it as well.

### CODE REVIEW

**Comment by WebDevBring** [comment]
[trello.com/c/TEi0FZCa](https://trello.com/c/TEi0FZCa)

File: css/algolia.css [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+  border-bottom-right-radius: 2px;
+  border-bottom-left-radius: 2px;
+  background-color: #fff;
+  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12);
+}
+
+html[data-theme=dark] .aa-Panel {
+  background-color: var(--green-15);
+  color: var(--green-86);
+}
```
Comment (Igor-J86) [inline css/algolia.css:20]: This part should be moved inside `dark-theme.css` as a nested class.

File: css/algolia.css [hunk 2]
```diff
@@ -88,6 +88,7 @@ html[data-theme=dark] .react-autosuggest__suggestion--highlighted {
   padding: 0.5rem 0.5rem 0.6rem 1rem;
   border-bottom: 1px solid hsl(0, 0%, 90%);
 }
+
 html[data-theme=dark] .hitlink {
   border-bottom-color: var(--green-darkest);
 }
```
Comment (Igor-J86) [inline css/algolia.css:94]: Should also go inside dark-theme.css as a nested class.

File: js/search/AlgoSearch.jsx [hunk 3]
```diff
@@ -1,36 +1,57 @@
 import algoliasearch from "algoliasearch/lite"
-import { InstantSearch } from "react-instantsearch-dom"
-import Result from "./Result.jsx"
+import AlgoLogo from "./AlgoLogo.jsx"
+import { InstantSearch } from "react-instantsearch";
+import { Autocomplete } from "./Autocomplete.jsx";
+import React from 'react';
```
Comment (Igor-J86) [inline js/search/AlgoSearch.jsx:5]: No need to import React here.

File: js/search/Autocomplete.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useRef, useState } from "react";
+import { render } from "react-dom";
+
+import { useSearchBox } from "react-instantsearch";
+import { autocomplete } from "@algolia/autocomplete-js";
+import { Fragment } from "react";
```
Comment (Igor-J86) [inline js/search/Autocomplete.jsx:6]: I think the imports can be combined into one line: 

```suggestion
import React,{ useEffect, useRef, useState, render, Fragment } from "react";
import { useSearchBox } from "react-instantsearch";
import { autocomplete } from "@algolia/autocomplete-js";
```

File: js/search/ResultItem.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
```
Comment (Igor-J86) [inline js/search/ResultItem.jsx:1]: No need to import React here either.

File: js/search/AlgoSearch.jsx [hunk 6]
```diff
@@ -1,36 +1,57 @@
...
+          },
+          getItems() {
+            return getAlgoliaResults({
+              searchClient,
+              queries: [
+                {
+                  indexName: __ALGOLIA_INDEX__,
+                  query,
+                  params: {
+                    hitsPerPage: 15,
```
[the remark this answers is not in the document]
Comment (e-kr) [reply js/search/AlgoSearch.jsx:36]: Good suggestion, looks better with 10 😄

File: js/search/Autocomplete.jsx [hunk 7]
```diff
@@ -1,9 +1,7 @@
-import React, { useEffect, useRef, useState } from "react";
+import React, { useEffect, useRef, useState, Fragment } from "react";
 import { render } from "react-dom";
```
Comment (Igor-J86) [inline js/search/Autocomplete.jsx:2]: I think `render` can also be imported from "react"..?

Comment (e-kr) [reply js/search/Autocomplete.jsx:2]: yep my bad 😅 will update!

**Review by Igor-J86 — APPROVED** [review]
Great job! Seems to be working fine with this approach! 🚀

**Pull request merged**